src/pages/storage/StorageVolumeForm.tsx

```diff
@@ -0,0 +1,298 @@
+  useNotify,
+} from "@canonical/react-components";
+import { useFormik } from "formik";
+import * as Yup from "yup";
+import { useQuery, useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import SubmitButton from "components/SubmitButton";
+import { createStorageVolume, fetchStoragePools } from "api/storage-pools";
+import { testDuplicateName } from "util/storage";
+import NotificationRow from "components/NotificationRow";
+import BaseLayout from "components/BaseLayout";
+import { useNavigate, useParams } from "react-router-dom";
+import StoragePoolSelector from "pages/storage/StoragePoolSelector";
+import { toBool } from "util/networkEdit";
+
+export interface StorageVolumeFormValues {
+  name: string;
+  pool: string;
+  size?: string;
+  security_shifted?: string;
+  security_unmapped?: string;
+  snapshots_expiry?: string;
+  snapshots_pattern?: string;
+  snapshots_schedule?: string;
+  block_filesystem?: string;
+  block_mount_options?: string;
+  lvm_stripes?: string;
+  lvm_stripes_size?: string;
+  zfs_blocksize?: string;
+  zfs_block_mode?: string;
+  zfs_delegate?: string;
+  zfs_remove_snapshots?: string;
+  zfs_use_refquota?: string;
+  zfs_reserve_space?: string;
+}
+
+const StorageVolumeForm: FC = () => {
+  const navigate = useNavigate();
+  const notify = useNotify();
+  const queryClient = useQueryClient();
+  const controllerState = useState<AbortController | null>(null);
+  const { project, pool } = useParams<{ project: string; pool: string }>();
+
+  if (!project) {
+    return <>Missing project</>;
+  }
+
+  if (!pool) {
+    return <>Missing storage pool name</>;
+  }
+
+  const { data: storagePools = [], error } = useQuery({
+    queryKey: [queryKeys.storage],
+    queryFn: () => fetchStoragePools(project),
+  });
+
+  if (error) {
+    notify.failure("Loading storage pools failed", error);
+  }
+
+  const StorageSchema = Yup.object().shape({
+    name: Yup.string()
+      .test(...testDuplicateName(project, controllerState))
+      .required("This field is required"),
+  });
+
+  const formik = useFormik<StorageVolumeFormValues>({
+    initialValues: {
+      name: "",
+      pool: pool,
+      size: "",
+    },
+    validationSchema: StorageSchema,
+    onSubmit: (values) => {
+      const volume = {
+        name: values.name,
+        config: {
+          size: values.size ? `${values.size}GiB` : undefined,
+          "security.shifted": toBool(values.security_shifted),
+          "security.unmapped": toBool(values.security_unmapped),
+          "snapshots.expiry": values.snapshots_expiry,
+          "snapshots.pattern": values.snapshots_pattern,
+          "snapshots.schedule": values.snapshots_schedule,
+          "block.filesystem": values.block_filesystem,
+          "block.mount_options": values.block_mount_options,
+          "lvm.stripes": values.lvm_stripes,
+          "lvm.stripes_size": values.lvm_stripes_size,
+          "zfs.blocksize": values.zfs_blocksize,
```

Comment:
Here too - the value should be sent as a string to avoid the unmarshal error:

![Screenshot from 2023-09-12 14:12:31](https://github.com/canonical/lxd-ui/assets/56583786/fa024f80-f34d-4d71-bb94-ea080d231ecf)

Also, I think it's unclear from the help text what the unit is. It would be better to specify that it is always KiB, so if you want to set it to 16 MiB you should not write 16 (which would cause an error), but 16384.